### ISSUE TITLE
rebase interactive: can't drop commit mid-rebase, when stopped on it

### ISSUE THREAD

**Issue by MikeSanMichel2** [magit/magit#2340]
When I have many commits included in the rebase, its difficult to figure out which I want removed
upfront at the "rebase-todo" screen. It's much easier to decide what changes I want later, in the status screen where I can see the contents of the diff. However, if I mark a commit for editing, realize
it is redundant and want to remove it I can't. "r e" has already moved on to the next commit,
and using `magit-reverse` + `commit extend` errors because git complains about `--allow-empty`.

I'd like to make all my decisions in the status buffer as I go along, with the diffs available, not up front
in "rebase-todo" where I can't see anything and have to both remember and plan all my decisions
in advance. It feels like doing the same thing twice.

continues #2330

**Comment by tarsius** [comment]
Do you want to keep an empty commit?

If so you can use <kbd>c - e e</kbd>.

Otherwise I would recommend moving to the previous commit and press either <kbd>x</kbd> or <kbd>C-u x</kbd> (not sure which you want).

**Comment by tarsius** [comment]
I am not quite sure I understood you correctly. <kbd>r s</kbd> might also be what you want.

**Comment by MikeSanMichel2** [comment]
<kbd>r s</kbd> looks like what I want, but it isn't working for me. Maybe I'm doing it wrong.

<kbd>r i</kbd>  on HEAD~
mark all for edit
<kbd>C-c C-c</kbd> 

in status screen the commit is marked "same"
<kbd>r s</kbd> 
the commit is marked "done".
<kbd>r s</kbd>  again (just to test)
rebase complete, back to log, all the commits are still there.

What am I missing?

**Comment by tarsius** [comment]
I am currently assuming that what you want is to drop the already stopped at commit:

```
Rebasing master onto master~2
pick 828950f magit-log-wash-rev: fix pasto
same 1ad2a96 magit-diff-wash-diffstat: prevent invalid hash
onto 761fe2d limit diff washing when creating log
```

In this case the stopped at commit is 1ad2a96. That commit currently also is the HEAD. So when I suggested to reset to HEAD~, I meant 761fe2d. Move there and press <kbd>x</kbd> with or without a prefix argument depending on what you want.

**Comment by MikeSanMichel2** [comment]
Using <kbd>- e</kbd> extendsfixes it though, of course. and I actually knew that. Sorry I'm just getting lost in all the buffer switching.

**Comment by MikeSanMichel2** [comment]
And yes, using "x" works too, but I seem to be constantly fighting magit and telling it not to do what it just did, so I can do what I'm trying to....

**Comment by MikeSanMichel2** [comment]
Much thanks for you help and patience.

**Comment by tarsius** [comment]
It does not look like rebase-i is going to make you happy. And I have no intentions to restore rewrite. So the best course of action might be if you extracted the feature from v1.4.2 and maintain it as a separate package.

**Comment by MikeSanMichel2** [comment]
I agree. I'll stop opening issues and scratch my own itch. Thanks again.

Final note, isn't https://github.com/magit/magit/issues/2340#issuecomment-148014593 a bug?

**Comment by tarsius** [comment]
It seems that `--skip` just falls back to `--continue` if the sequence was not stopped due to a conflict. So while that is a bit strange, I wouldn't consider it a bug (and if it is a bug then it is in git).

**Comment by MikeSanMichel2** [comment]
I've downgraded to magit 1.4.2, I find rewrite more intuitive but I have to admit that some
of the things I've pointed out were warts in 1.4.2 as well. Still, there's no "rebase-todo"
step, and It simply gives me an empty base and a commit queue to work through, which I prefer.

If I get the time and commitment, I'll try to modernize this into a companion package for v2.x.
I hope you'll be able to spare the time to provide some guidance/review when the time comes.

I'm sure we'll meet again...
